### PR TITLE
feat(radios): upgrade radios - INNO-631

### PR DESCRIPTION
**Pull request by yhuard**
TODO: fix the errors reported by axe-core:

```
- [critical] Ensures related <input type="radio"> elements have a group and that the group designation is consistent: -- [critical] #opt-1: --- [critical] All elements with the name "radio-example" do not reference the same element with aria-labelledby--- [critical] Element does not have a containing fieldset or ARIA group
```

### CODE REVIEW

File: framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.scss [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+.ecl-radio {
+  line-height: 2;
+  margin: 0;
+}
+
+.ecl-radio__label {
+  display: inline-block;
+  padding-left: 34px;
```
Comment (emeryro) [inline framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.scss]: Magic number..
We should use same spacing as checkboxes.

File: framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.twig [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-  <div class="radio">
-    <input type="radio" id="rd-example-3" value="" checked="checked">
-    <label for="rd-example-3">Checked multiple option 3</label>
-  </div>
-  <p class="form-field-feedback">Some feedback message</p>
+{% set id = name|default('ecl-radio-1') %}
+{% set name = name|default('ecl-radio') %}
+{% set desc = desc|default('Description') %}
+
+<div role="radiogroup" aria-labelledby="{{ id }}_label" id="{{ id }}">
```
Comment (emeryro) [inline framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.twig]: Missing extraClass and extraAttributes

File: framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.twig [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+  <p id="{{ id }}_label" class="ecl-sr-only">{{ desc }}</p>
+  {% for option in options %}
+    <div class="ecl-radio">
+      <label for={{ option.id }}>
+        <input class="ecl-radio__input ecl-sr-only" type="radio" id="{{ option.id }}" value="{{ option.value }}" name="{{ name }}"{{ option.checked ? ' checked' : '' }} />
+        <span class="ecl-radio__label">{{ option.label }}</span>
+      </label>
+    </div>
+  {% endfor %}
 </div>
```
Comment (emeryro) [inline framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.twig]: I suggest to add disabled and error states too (as for other form elements)

File: framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.scss [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    background: #fff url('../images/svg/radio-unchecked.svg') no-repeat;
+    border: 3px solid transparent;
+    border-radius: 50%;
+    content: '';
+    display: inline-block;
+    height: 1.5em;
+    left: 0;
+    position: absolute;
+    top: 50%;
+    transform: translateY(-50%);
```
Comment (emeryro) [inline framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.scss]: I'm not a big fan of these "transform hack"..
In checkboxes component it has been done using line-height.

**Review by emeryro — CHANGES_REQUESTED** [review]
A few modifications requested, but overall display is good

File: framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.scss (hunk 1 above)
Comment (yhuard) [inline framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.scss]: Totally agree!

File: framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.scss (hunk 4 above)
Comment (yhuard) [inline framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.scss]: And what happens when the label takes 2 lines? IMHO line-height is worse than transform

File: framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.twig (hunk 3 above)
Comment (yhuard) [inline framework/components/ecl-forms/ecl-forms-radios/ecl-forms-radios.twig]: Will do

**Review by kalinchernev — APPROVED** [review]

**Review by emeryro — APPROVED** [review]

**Pull request merged**